### ISSUE TITLE
19.5.3.8 (19.4.4.33+) segmentation fault on CH-client query to system catalog.

### ISSUE THREAD

**Issue by den-crane** [yandex/ClickHouse#5066]
I experienced several crashes right after I ran create table from CH client.
Probably the cause is the CH cli 'auto-complete' query:

SELECT DISTINCT arrayJoin(extractAll(name, '[\\w_]{2,}')) AS res FROM (SELECT name FROM sy
stem.functions UNION ALL SELECT name FROM system.table_engines UNION ALL SELECT name FROM system.formats UNION ALL SELECT name FROM system.table_functions UNION ALL SELECT name FROM system.data_type_families UNION ALL SELECT name FROM system.settings UNION ALL SELECT concat(func.name, comb.name) FROM system.functions AS func CROSS JOIN system.aggregate_function_combinators AS comb WHERE is_aggregate UNION
 ALL SELECT name FROM system.databases LIMIT 10000 UNION ALL SELECT DISTINCT name FROM system.tables LIMIT 10000 UNION ALL SELECT DISTINCT name FROM system.columns LIMIT 10000) WHERE notEmpty(res)

[19_5_3_8_Segmentation_fault.txt](https://github.com/yandex/ClickHouse/files/3100307/19_5_3_8_Segmentation_fault.txt)

**Comment by den-crane** [comment]
The same with 19.7.1.360

```
./crash.sh
1
2
3
4
5
6
7
Error on processing query: SELECT DISTINCT arrayJoin(extractAll(name, '[\w_]{2,}')) AS res FROM (SELECT name FROM system.functions UNION ALL SELECT name FROM system.table_engines UNION ALL SELECT name FROM system.formats UNION ALL SELECT name FROM system.table_functions UNION ALL SELECT name FROM system.data_type_families UNION ALL SELECT name FROM system.settings UNION ALL SELECT concat(func.name, comb.name) FROM system.functions AS func CROSS JOIN system.aggregate_function_combinators AS comb WHERE is_aggregate UNION
	ALL SELECT name FROM system.databases LIMIT 10000 UNION ALL SELECT DISTINCT name FROM system.tables LIMIT 10000 UNION ALL SELECT DISTINCT name FROM system.columns LIMIT 10000) WHERE notEmpty(res)
format Null;
;

Code: 32, e.displayText() = DB::Exception: Attempt to read after eof: while receiving packet from localhost:9000, ::1, Stack trace:

0. clickhouse-client(StackTrace::StackTrace()+0x16) [0x7678a76]
1. clickhouse-client(DB::Exception::Exception(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, int)+0x22) [0x3927652]
0. clickhouse-client(StackTrace::StackTrace()+0x16) [0x7678a76]
1. clickhouse-client(DB::Exception::Exception(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, int)+0x22) [0x3927652]
2. clickhouse-client(DB::throwReadAfterEOF()+0x4d) [0x39276bd]
3. clickhouse-client(DB::Connection::receivePacket()+0xf35) [0x6e92455]
4. clickhouse-client(DB::Client::receiveAndProcessPacket()+0x3c) [0x3996c4c]
5. clickhouse-client(DB::Client::receiveResult()+0x164) [0x3997f94]
6. clickhouse-client(DB::Client::processSingleQuery(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, std::shared_ptr<DB::IAST>)+0x320) [0x3999410]
7. clickhouse-client(DB::Client::process(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&)+0x322) [0x399b7e2]
8. clickhouse-client(DB::Client::mainImpl()+0x1278) [0x399e1e8]
9. clickhouse-client(DB::Client::main(std::vector<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, std::allocator<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > > > const&)+0x25) [0x387deb5]
10. clickhouse-client(Poco::Util::Application::run()+0x35) [0x7793ad5]
11. clickhouse-client(mainEntryClickHouseClient(int, char**)+0x46) [0x394ca16]
12. clickhouse-client(main+0x1e8) [0x3877a78]
13. /lib/x86_64-linux-gnu/libc.so.6(__libc_start_main+0xf1) [0x7ff92c2dc2e1]
14. clickhouse-client(_start+0x2a) [0x3923eaa]
 (version 19.7.1.360 (official build))8
Code: 210. DB::NetException: Connection refused (localhost:9000, ::1)



```


cat crash.sql
```
SELECT DISTINCT arrayJoin(extractAll(name, '[\w_]{2,}')) AS res FROM (SELECT name FROM system.functions UNION ALL SELECT name FROM system.table_engines UNION ALL SELECT name FROM system.formats UNION ALL SELECT name FROM system.table_functions UNION ALL SELECT name FROM system.data_type_families UNION ALL SELECT name FROM system.settings UNION ALL SELECT concat(func.name, comb.name) FROM system.functions AS func CROSS JOIN system.aggregate_function_combinators AS comb WHERE is_aggregate UNION
	ALL SELECT name FROM system.databases LIMIT 10000 UNION ALL SELECT DISTINCT name FROM system.tables LIMIT 10000 UNION ALL SELECT DISTINCT name FROM system.columns LIMIT 10000) WHERE notEmpty(res)
format Null;
;
```

cat crash.sh
```
#!/bin/bash

for i in `seq 1 10`;
 do
         echo $i
         cat crash.sql | clickhouse-client -mn --enable_optimize_predicate_expression 0 --allow_experimental_multiple_joins_emulation 0 --allow_experimental_cross_to_join_conversion 0
 done
```

**Comment by den-crane** [comment]
select name,value from system.settings where changed;
```

┌─name──────────────────────────────────────┬─value────────────┐
│ max_query_size                            │ 41943040         │
│ connect_timeout_with_failover_ms          │ 1000             │
│ use_uncompressed_cache                    │ 1                │
│ optimize_move_to_prewhere                 │ 0                │
│ load_balancing                            │ nearest_hostname │
│ distributed_aggregation_memory_efficient  │ 1                │
│ skip_unavailable_shards                   │ 1                │
│ http_connection_timeout                   │ 15               │
│ empty_result_for_aggregation_by_empty_set │ 1                │
│ max_bytes_before_external_group_by        │ 25374640128      │
│ max_memory_usage                          │ 50749280256      │
└───────────────────────────────────────────┴──────────────────┘
```

**Comment by den-crane** [comment]
The minimal 

cat crash.sql
```
SELECT name FROM system.tables
UNION ALL
SELECT name FROM system.columns
format Null;
```

**Comment by alexey-milovidov** [comment]
https://github.com/yandex/ClickHouse/pull/4535/commits/0d7ab9e3ea25d099f55b0e2c1e5e1f736bae6260

**Comment by abyss7** [comment]
Found the culprit code - trying to fix.

**Comment by den-crane** [comment]
fixed in 19.6.2.11-stable
